runtime/src/lib.rs

```diff
@@ -52,9 +55,11 @@ impl ffi::IRDescriptorTableEntry {
     #[doc(alias = "IRDescriptorTableSetBufferView")]
     pub fn buffer_view(buffer_view: &BufferView<'_>) -> Self {
         Self {
-            gpuVA: buffer_view.buffer.gpu_address() + buffer_view.buffer_offset,
+            gpuVA: buffer_view.buffer.gpuAddress() + buffer_view.buffer_offset,
             textureViewID: match buffer_view.texture_buffer_view {
-                Some(texture) => texture.gpu_resource_id()._impl,
+                Some(texture) => unsafe {
+                    std::mem::transmute::<objc2_metal::MTLResourceID, u64>(texture.gpuResourceID())
+                },
                 None => 0,
```

Comment:
I can't find the implementation of `ffi::IRDescriptorTableEntry`, but I'm guessing it's not possible to make `textureViewID` a `MTLResourceID`?

Reply:
[It's available in this repo]https://github.com/Traverse-Research/saxaboom/blob/74e8dc560cee5f43d948c2297e440db8af969221/bindings_generator/vendor/metal_irconverter_runtime/metal_irconverter_runtime.h#L666)

The reason we need a transmute is that objc2-metal doesn't provide a way to get the u64 out that we need to pass to the FFI.

Reply:
Thanks, I searched for it in the docs pages. Seems they are in binding files so it's good then!

Reply:
The headers for IRConverter do use these `MTLResourceID` types in some _but not all_ places, this could perhaps be reported upstream.